flake.nix

```diff
@@ -0,0 +1,55 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/master";
+    flake-utils.url = "github:numtide/flake-utils";
+    gomod2nix = {
+      url = "github:nix-community/gomod2nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+      inputs.utils.follows = "flake-utils";
+    };
+  };
+
+  outputs = { self, nixpkgs, gomod2nix, flake-utils }:
+    {
+      overlays.default = pkgs: _: {
+        simd = pkgs.callPackage ./simapp { rev = self.shortRev or "dev"; };
+      };
+    } //
+    (flake-utils.lib.eachDefaultSystem
+      (system:
+        let
+          mkApp = drv: {
+            type = "app";
+            program = "${drv}/bin/${drv.meta.mainProgram}";
+          };
+          pkgs = import nixpkgs {
+            inherit system;
+            config = { };
+            overlays = [
+              gomod2nix.overlays.default
+              self.overlays.default
+            ];
+          };
+        in
+        rec {
+          packages = rec {
+            default = simd;
+            inherit (pkgs) simd;
+          };
+          apps = rec {
+            default = simd;
+            simd = mkApp pkgs.simd;
+          };
+          devShells = rec {
+            default = simd;
+            simd = with pkgs; mkShell {
+              buildInputs = [
+                go_1_20 # Use Go 1.20 version
+                rocksdb
```

Comment:
How do we specifiy which version of the librocksdb we want?

Reply:
There are several ways:
1. Pin to a nixpkgs version that contains the desired rocksdb version, that happens to be the case right now
2. Override the src tarball in rocksdb derivative, for example in the overlay:
```
rocksdb = prev.rocksdb.overrides {
  version = xxx;
  src = fetchGithub {
    repo
    hash
    etc..
  };
};
```